wireapp/wire-android
# feat: legal hold indication on user profiles [WPB-4780]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4780" title="WPB-4780" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4780</a>  [Android] Legal hold indication on user profiles
  </td></table>
  <br />
 
















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

As a user I want to see an indication on user profiles that are under legal hold so that I’m more aware of my privacy.

### Solutions

Change the legal hold banner to distinguish between "legal hold active" and "legal hold request pending".
Add legal hold banner to self user profile in both cases and to other user profile when active.
Show proper dialogs when clicking on legal hold banner.

### Testing

#### Test Coverage (Optional)


#### How to Test

Request legal hold for the current user or enable it for other user and open the self or other user profile to see the banner.

### Attachments (Optional)

<img src="https://github.com/wireapp/wire-android/assets/30429749/89e63d86-8441-4651-9461-201e91bfd391" width=400/>



<img src="https://github.com/wireapp/wire-android/assets/30429749/f416c8b3-96bd-4ed3-b5ef-ca7c4e744ed2" width=400/>


https://github.com/wireapp/wire-android/assets/30429749/7dd62048-ff7b-47b2-8bee-64279174af9d

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
